mrtt-ui/src/components/CheckboxGroup.js

```diff
@@ -0,0 +1,61 @@
+import { Checkbox, FormControlLabel, FormGroup } from '@mui/material'
+import PropTypes from 'prop-types'
+import { forwardRef, useEffect, useState } from 'react'
+
+const CheckboxGroup = forwardRef(({ options, value, onChange, onBlur }, ref) => {
+  const [checkboxItems, setCheckboxItems] = useState([])
+
+  const _loadCheckboxItems = useEffect(() => {
+    setCheckboxItems(value)
+  }, [value])
+
+  const handleCheckboxGroupChange = ({ itemValue, event }) => {
+    const updateCheckboxItems = [...checkboxItems]
+    const foundItemIndex = updateCheckboxItems.indexOf(itemValue)
+
+    if (foundItemIndex > -1) {
+      updateCheckboxItems.splice(foundItemIndex, 1)
+    } else {
+      updateCheckboxItems.push(itemValue)
+    }
+
+    setCheckboxItems(updateCheckboxItems)
+    onChange({ selectedItems: updateCheckboxItems, event })
+  }
+
+  const checkboxes = options.map((item) => (
+    <div key={item.value}>
+      <FormControlLabel
+        control={
+          <Checkbox
+            id={item.value}
+            value={item.value}
+            checked={checkboxItems.includes(item.value)}
+            onChange={(event) => handleCheckboxGroupChange({ itemValue: item.value, event })}
+            onBlur={onBlur}
+          />
+        }
+        label={item.label}
+      />
+    </div>
+  ))
+
+  return <FormGroup ref={ref}>{checkboxes}</FormGroup>
+})
+
+CheckboxGroup.displayName = 'Checkbox Group'
+CheckboxGroup.propTypes = {
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      label: PropTypes.string,
+      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+    })
+  ).isRequired,
+  onChange: PropTypes.func.isRequired,
+  onBlur: PropTypes.func,
+  value: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number]))
+}
+
```

Comment:
This component is just a generic unopinionated (other than MUI) checkbox  group.

Reply:
similar question to before, why are we setting checkbox items? this would be handled within react hook form

Reply:
maybe we can hop on a call to discuss! did something quite different today with the `siteBackgroundForm`

Reply:
We're treating the checkbox group as one input, so react hook form doesnt need to concern itself with each individual checkbox input. Most of my day is free for a call if you want.